### ISSUE TITLE
Unwinding assertion failures do not clearly indicate which loop is involved

### ISSUE THREAD

**Issue by tedinski** [model-checking/kani#1345]
```
Check 67: .unwind.0
         - Status: FAILURE
         - Description: "unwinding assertion loop 0"


SUMMARY: 
 ** 1 of 67 failed (66 undetermined)
Failed Checks: unwinding assertion loop 0
```

What function? What loop?

I vaguely recall we used to see that information, so something changed.

Raw cbmc output:

```
** Results:
[.unwind.0] unwinding assertion loop 0: FAILURE
```

**Comment by tedinski** [comment]
Probably a bug on our side, I did a quick test with cbmc:

```
** Results:
test.c function main
[main.unwind.0] line 3 unwinding assertion loop 0: FAILURE
```

**Comment by zhassan-aws** [comment]
This information is typically included, e.g for:
```rust
#[kani::proof]
#[kani::unwind(5)]
fn harness() {
    let mut counter = 0;
    loop {
        counter += 1;
        assert!(counter < 10);
    }
}
```
the following is reported:
```bash
$ kani test.rs 
Kani Rust Verifier 0.66.0 (standalone)
Checking harness harness...
CBMC 6.7.1 (cbmc-6.7.1)
CBMC version 6.7.1 (cbmc-6.7.1) 64-bit x86_64 linux
Reading GOTO program from file /home/ubuntu/git/kani/tests/kani/Unwind-Attribute/test__RNvCs9rQ4aHrvVY3_4test7harness.out
Generating GOTO Program
Adding CPROVER library (x86_64)
Removal of function pointers and virtual functions
Generic Property Instrumentation
Running with 16 object bits, 48 offset bits (user-specified)
Starting Bounded Model Checking
Unwinding loop _RNvCs9rQ4aHrvVY3_4test7harness.0 iteration 1 file test.rs line 5 column 5 function harness thread 0
Unwinding loop _RNvCs9rQ4aHrvVY3_4test7harness.0 iteration 2 file test.rs line 5 column 5 function harness thread 0
Unwinding loop _RNvCs9rQ4aHrvVY3_4test7harness.0 iteration 3 file test.rs line 5 column 5 function harness thread 0
Unwinding loop _RNvCs9rQ4aHrvVY3_4test7harness.0 iteration 4 file test.rs line 5 column 5 function harness thread 0
Not unwinding loop _RNvCs9rQ4aHrvVY3_4test7harness.0 iteration 5 file test.rs line 5 column 5 function harness thread 0
Runtime Symex: 0.00410914s
size of program expression: 182 steps
slicing removed 128 assignments
Generated 21 VCC(s), 11 remaining after simplification
Runtime Postprocess Equation: 2.5891e-05s
Passing problem to propositional reduction
converting SSA
Runtime Convert SSA: 8.8843e-05s
Running propositional reduction
Post-processing
Runtime Post-process: 1.101e-05s
Solving with CaDiCaL 2.0.0
1 variables, 0 clauses
SAT checker: instance is SATISFIABLE
Runtime Solver: 5.7502e-05s
Runtime decision procedure: 0.000189535s

RESULTS:
Check 1: harness.assertion.1
         - Status: UNDETERMINED
         - Description: "attempt to add with overflow"
         - Location: test.rs:6:9 in function harness

Check 2: harness.assertion.2
         - Status: UNDETERMINED
         - Description: "assertion failed: counter < 10"
         - Location: test.rs:7:9 in function harness

Check 3: harness.unwind.0
         - Status: FAILURE
         - Description: "unwinding assertion loop 0"
         - Location: test.rs:5:5 in function harness


SUMMARY:
 ** 1 of 3 failed (2 undetermined)
Failed Checks: unwinding assertion loop 0
 File: "test.rs", line 5, in harness

VERIFICATION:- FAILED
[Kani] info: Verification output shows one or more unwinding failures.
[Kani] tip: Consider increasing the unwinding value or disabling `--unwinding-assertions`.
Verification Time: 0.030533027s

Manual Harness Summary:
Verification failed for - harness
Complete - 0 successfully verified harnesses, 1 failures, 1 total.
```
Closing.
